ibm/resource_ibm_cis_certificate_order.go

```diff
@@ -0,0 +1,198 @@
+package ibm
+
+import (
+	"log"
+	"time"
+
+	"github.com/IBM/go-sdk-core/v3/core"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+const (
+	cisCertificateOrderID            = "certificate_id"
+	cisCertificateOrderHosts         = "hosts"
+	cisCertificateOrderType          = "type"
+	cisCertificateOrderTypeDedicated = "dedicated"
+	cisCertificateOrderStatus        = "status"
+	cisCertificateOrderDeleted       = "deleted"
+	cisCertificateOrderDeletePending = "deleting"
+)
+
+func resourceIBMCISCertificateOrder() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMCISCertificateOrderCreate,
+		Update:   resourceIBMCISCertificateOrderRead,
+		Read:     resourceIBMCISCertificateOrderRead,
+		Delete:   resourceIBMCISCertificateOrderDelete,
+		Exists:   resourceIBMCISCertificateOrderExist,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS object id or CRN",
+				Required:    true,
+			},
+			cisDomainID: {
+				Type:        schema.TypeString,
+				Description: "Associated CIS domain",
+				Required:    true,
+			},
+			cisCertificateOrderID: {
+				Type:        schema.TypeString,
+				Description: "certificate id",
+				Computed:    true,
+			},
+			cisCertificateOrderHosts: {
+				Type:        schema.TypeList,
+				Description: "Hosts which certificate need to be ordered",
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			cisCertificateOrderStatus: {
+				Type:        schema.TypeString,
+				Description: "certificate status",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceIBMCISCertificateOrderCreate(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisSSLClientSession()
+	if err != nil {
+		return err
+	}
+	crn := d.Get(cisID).(string)
+	zoneID := d.Get(cisDomainID).(string)
+	cisClient.Crn = core.StringPtr(crn)
+	cisClient.ZoneIdentifier = core.StringPtr(zoneID)
+
+	hosts := d.Get(cisCertificateOrderHosts)
+	hostsList := expandStringList(hosts.([]interface{}))
+	opt := cisClient.NewOrderCertificateOptions()
+	opt.SetType(cisCertificateOrderTypeDedicated)
+	opt.SetHosts(hostsList)
+
+	result, resp, err := cisClient.OrderCertificate(opt)
```

Comment:
Is wait logic for ordering certificate not required?

Reply:
no need. Order certificate will be done instantly.